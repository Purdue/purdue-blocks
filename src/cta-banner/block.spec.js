jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickRadio,
} from '../test-helpers'

const block = {blockTitle: 'CTA Banner', blockName: 'purdue-blocks/cta-banner'}

describe( 'ðŸ”¬ CTA Banner Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock( block.blockTitle )

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe('ðŸ”¬ Block Editor Fields', () => {
        test( 'ðŸ”Ž CTA Description Text Field', async () => {
            await blockStartup(block)

            const typeString = "CTA Description Text test."

            await page.focus('input[placeholder="CTA Description Text..."]')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"ctaDesc":"${typeString}"`)).toBe(true)
        })
        test( 'ðŸ”Ž CTA Button Text Field', async () => {
            await blockStartup(block)

            const typeString = "CTA Button test."

            await page.focus('input[placeholder="CTA Button Text..."]')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"ctaText":"${typeString}"`)).toBe(true)
        })
        test( 'ðŸ”Ž CTA Link URL Field', async () => {
            await blockStartup(block)

            const typeString = "https://www.purdue.edu"

            await page.focus('input[placeholder="CTA URL..."]')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"ctaUrl":"${typeString}"`)).toBe(true)
        })
    })

    describe('ðŸ”¬ Side Panel Settings', () => {
        test( 'ðŸ”Ž Banner type radio controls', async () => {
            await blockStartup(block)

            await clickRadio('CTA Banner Type', 'black')
            
            let editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"type":"black"`)).toBe(true)


            await clickRadio('CTA Banner Type', 'gold')
            
            editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"type":"black"`)).toBe(false)
            expect( editedContent.includes(`"type":"gray"`)).toBe(false)
            expect( editedContent.includes(`"type":"image"`)).toBe(false)


            await clickRadio('CTA Banner Type', 'gray')
            
            editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"type":"gray"`)).toBe(true)


            await clickRadio('CTA Banner Type', 'image')
            
            editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"type":"image"`)).toBe(true)
        })
        test('ðŸ”Ž Image banner type media library button', async () => {
            await blockStartup(block)

            await clickRadio('CTA Banner Type', 'image')
            
            const editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"type":"image"`)).toBe(true)

            // open media library
            await clickElementByText('button', 'Open Media Library')

            // tests that media library does open
            expect(await page.$(`div.media-frame-title`)).not.toBeNull()
        })
        test('ðŸ”Ž Image banner type alt text field', async () => {
            await blockStartup(block)
            
            await clickRadio('CTA Banner Type', 'image')
            let editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"type":"image"`)).toBe(true)

            const typeString = "Image alt text."
            
            // focus the text box then type into it with the virtual keyboard
            await clickElementByText('label', 'Hero Image Alt Text')
            await page.keyboard.type(typeString, {delay: 10})
    
            editedContent = await getEditedPostContent()

            expect( editedContent).toMatchSnapshot()
            expect( editedContent.includes(`aria-label="${typeString}"`)).toBe(true)
            expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
        })
    })
})