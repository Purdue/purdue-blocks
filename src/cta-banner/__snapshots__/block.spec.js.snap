// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 CTA Banner Block 🔎 Block should be available. 1`] = `
"<!-- wp:purdue-blocks/cta-banner -->
<div class=\\"pu-cta-banner pu-cta-banner-gold
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-gold__desc\\"></p><a href=\\"\\" class=\\"pu-cta-banner-gold__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Block Editor Fields 🔎 CTA Button Text Field 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"ctaText\\":\\"CTA Button test.\\"} -->
<div class=\\"pu-cta-banner pu-cta-banner-gold
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-gold__desc\\"></p><a href=\\"\\" class=\\"pu-cta-banner-gold__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\">CTA Button test.</a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Block Editor Fields 🔎 CTA Description Text Field 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"ctaDesc\\":\\"CTA Description Text test.\\"} -->
<div class=\\"pu-cta-banner pu-cta-banner-gold
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-gold__desc\\">CTA Description Text test.</p><a href=\\"\\" class=\\"pu-cta-banner-gold__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Block Editor Fields 🔎 CTA Link URL Field 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"ctaUrl\\":\\"https://www.purdue.edu\\"} -->
<div class=\\"pu-cta-banner pu-cta-banner-gold
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-gold__desc\\"></p><a href=\\"https://www.purdue.edu\\" class=\\"pu-cta-banner-gold__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Banner type radio controls 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"type\\":\\"black\\"} -->
<div class=\\"pu-cta-banner pu-cta-banner-black
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-black__desc\\"></p><a href=\\"\\" class=\\"pu-cta-banner-black__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Banner type radio controls 2`] = `
"<!-- wp:purdue-blocks/cta-banner -->
<div class=\\"pu-cta-banner pu-cta-banner-gold
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-gold__desc\\"></p><a href=\\"\\" class=\\"pu-cta-banner-gold__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Banner type radio controls 3`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"type\\":\\"gray\\"} -->
<div class=\\"pu-cta-banner
       pu-cta-banner-gray\\" aria-label=\\"\\"><div class=\\"container\\"><a href=\\"\\" class=\\"pu-cta-banner-gray__desc\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Banner type radio controls 4`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"type\\":\\"image\\"} -->
<div class=\\"pu-cta-banner
       pu-cta-banner-image\\" aria-label=\\"\\"><div class=\\"container\\"><a href=\\"\\" class=\\"pu-cta-banner-image__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Image banner type alt text field 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"type\\":\\"image\\"} -->
<div class=\\"pu-cta-banner
       pu-cta-banner-image\\" aria-label=\\"\\"><div class=\\"container\\"><a href=\\"\\" class=\\"pu-cta-banner-image__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Image banner type alt text field 2`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"type\\":\\"image\\",\\"altText\\":\\"Image alt text.\\"} -->
<div class=\\"pu-cta-banner
       pu-cta-banner-image\\" aria-label=\\"Image alt text.\\"><div class=\\"container\\"><a href=\\"\\" class=\\"pu-cta-banner-image__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Image banner type media library button 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"type\\":\\"image\\"} -->
<div class=\\"pu-cta-banner
       pu-cta-banner-image\\" aria-label=\\"\\"><div class=\\"container\\"><a href=\\"\\" class=\\"pu-cta-banner-image__button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;

exports[`🔬 CTA Banner Block 🔬 Side Panel Settings 🔎 Link can be opened in new tab 1`] = `
"<!-- wp:purdue-blocks/cta-banner {\\"external\\":true} -->
<div class=\\"pu-cta-banner pu-cta-banner-gold
      \\" aria-label=\\"\\"><div class=\\"container\\"><p class=\\"pu-cta-banner-gold__desc\\"></p><a href=\\"\\" class=\\"pu-cta-banner-gold__button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"></a></div></div>
<!-- /wp:purdue-blocks/cta-banner -->"
`;
